libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/model/RouteShield.kt

```diff
@@ -201,11 +200,7 @@ sealed class RouteShield(
          * @param other mapbox designed shield
          */
         fun compareWith(other: MapboxShield?): Boolean {
```

Comment:
Why don't we deprecate the suggest developers just use `equals` instead?

Reply:
We have a similar function in `MapboxLegacyShield` class. 
I think these functions are useful in the sense that they showcase what data is needed to identify necessary shield. 
But this data can probably change in the future, so in my opinion we shouldn't have made these functions public.

Reply:
Deprecated these functions in a separate commit, but would also like to hear @abhishek1508's opinion.

Reply:
Looking at the current implementation again, these functions were made public because currently they are hosted in `libnavui-shield` module. However, banner instructions in the `libnavui-maneuver` module also use these functions to show the shields. Hence they cannot be `internal`. I also don't think the data will change since this data is directly coming from `DirectionsResponse` and it's highly unlikely that any data structure would change in the route response unless critically wrong.

I would also not agree for deprecation, since even though they are used internally in the SDK to make comparisons but the fact they are public, it doesn't look good on us to deprecate right away since this feature was just released in `v2.2.0`

/cc @Zayankovsky @LukasPaczos

Reply:
This function (only the one for the `MapboxDesignShield`, not the legacy one) is now on par with a simple equality function 
 so I don't see why we'd want ot keep using it and advertise it. We should just migrate to equality check.

We may do that separately though, it's more of a nit.

Reply:
Yep I am not against the fact that it is same as `equals`. I am just reluctant to deprecate it right away since our latest stable is still `v2.2` and I feel the timing is not correct.